### PR TITLE
Test that time passes between func and ext. return

### PR DESCRIPTION
**Pull request by imphil**
This test was developed as part of https://github.com/potentialventures/cocotb/pull/652 but works now without the fix in this PR. Let's take it to simplify the PR.

### CODE REVIEW

**Comment by imphil** [comment]
@eric-wieser could you have a look at this one?

File: tests/test_cases/test_external/test_external.py [hunk 1]
```diff
@@ -205,6 +205,30 @@ def test_time_in_external(dut):
...
+
+
+@cocotb.test(expect_fail=False)
+def test_instantaneous_return(dut):
+    """Test that the simulation time does not advance between the final
+    function call in the external and the calling coroutine continuing"""
+    period = 100
+    clk_gen = cocotb.fork(Clock(dut.clk, period).start())
+    yield Timer(10, 'ns')
+    ext_sim_time = yield cocotb.external(test_ext_instantaneous_return)(dut)
```
Comment (eric-wieser) [inline tests/test_cases/test_external/test_external.py]: I think you maybe just want a `cocotb.external` decorator on `test_ext_instantaneous_return`.

File: tests/test_cases/test_external/test_external.py [hunk 2]
```diff
@@ -205,6 +205,30 @@ def test_time_in_external(dut):
...
+def yield_to_next_clock(dut):
+    yield RisingEdge(dut.clk)
+
+
+def test_ext_instantaneous_return(dut):
+    yield_to_next_clock(dut)
+    return get_sim_time('ps')
+
+
+@cocotb.test(expect_fail=False)
```
Comment (eric-wieser) [inline tests/test_cases/test_external/test_external.py]: `expect_fail=False` is the default, there is no need to have this here.

**Comment by eric-wieser** [comment]
I think the test here is now a duplicate of the one merged in #1171

**Pull request closed without merging**